### PR TITLE
nixos/modules: services.syncthing add guiAddress parameter

### PR DESCRIPTION
**Pull request by mrVanDalo**
###### Motivation for this change

To make it possible to configure your syncthing from a remote machine.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/networking/syncthing.nix [hunk 1]
```diff
@@ -16,6 +16,14 @@ in {
         available on http://127.0.0.1:8384/.
       '';
 
+      guiAddress = mkOption {
+        type = types.string;
```
Comment (Infinisil) [inline nixos/modules/services/networking/syncthing.nix]: Use `types.str` instead

File: nixos/modules/services/networking/syncthing.nix [hunk 2]
```diff
@@ -16,6 +16,14 @@ in {
...
+        default = "127.0.0.1:8384";
+        description = ''
+          Override GUI listen address.
```
Comment (Infinisil) [inline nixos/modules/services/networking/syncthing.nix]: How about "Address to server the GUI on" instead? The fact that it overrides the default shouldn't be in the options description

**Review by infinisil — CHANGES_REQUESTED** [review]

File: nixos/modules/services/networking/syncthing.nix (hunk 1 above)
Comment (mrVanDalo) [reply nixos/modules/services/networking/syncthing.nix]: I used types.string for consistency ( it is used in the other options as well )

File: nixos/modules/services/networking/syncthing.nix (hunk 2 above)
Comment (mrVanDalo) [reply nixos/modules/services/networking/syncthing.nix]: Done

File: nixos/modules/services/networking/syncthing.nix (hunk 1 above)
Comment (Infinisil) [reply nixos/modules/services/networking/syncthing.nix]: Change all of them to `str` then :)

**Review by peterhoeg — CHANGES_REQUESTED** [review]
Looks good once @Infinisil 's concerns have been addressed.

File: nixos/modules/services/networking/syncthing.nix (hunk 1 above)
Comment (mrVanDalo) [reply nixos/modules/services/networking/syncthing.nix]: Done

**Review by peterhoeg — APPROVED** [review]

**Comment by peterhoeg** [comment]
Looks good, thank you for your contribution @mrVanDalo

**Comment by matthewbauer** [comment]
Could you add a release note that explains guiAddress change? This appears to override the default from configDir.

https://github.com/NixOS/nixpkgs/issues/49805

**Comment by peterhoeg** [comment]
All, please note that the latest version .52 is just out which supports a "unix://" parameter to guiAddress - would be neat to support that here as part of this PR.

**Pull request merged**